src-java/rule-manager/rule-manager-implementation/src/main/java/org/openkilda/rulemanager/RuleManagerImpl.java

```diff
@@ -612,19 +631,24 @@ private List<SpeakerData> buildReverseHaRules(
         boolean sharedMeterWasCreated = false;
         boolean sharedPathRulesWereCreated = false;
 
+        Set<PathId> subPathIds = haPath.getSubPathIds();
         for (FlowPath subPath : haPath.getSubPaths()) {
 
             // reverse ingress rule
             if (ingress) {
                 Set<FlowSideAdapter> overlappingAdapters = new HashSet<>();
                 if (filterOutUsedSharedRules) {
-                    overlappingAdapters.addAll(getOverlappingMultiTableIngressAdapters(haFlow, subPath, adapter));
+                    overlappingAdapters.addAll(getOverlappingMultiTableIngressAdapters(
+                            haFlow, subPath, subPathIds, adapter));
                 }
                 if (subPath.getSrcSwitchId().equals(haPath.getYPointSwitchId())) {
+                    boolean generateMeterCommand = generateReverseIngressMeter(
```

Comment:
If the previous comment regarding method name is counted, then imho it is better to change this variable as well. 

```suggestion
                    boolean shouldGenerateMeterCommand = generateReverseIngressMeter(
```

Reply:
fixed